_source/_posts/2019-12-12-tutorial-kotlin-beginners-guide.md

```diff
@@ -0,0 +1,491 @@
+
+Using the [Spring Initializr API](https://start.spring.io/), create a Maven project with the following command:
+
+```shell
+curl https://start.spring.io/starter.zip -d dependencies=web,okta \
+-d language=kotlin \
+-d type=maven-project \
+-d groupId=com.okta.developer \
+-d artifactId=kotlin-spring-boot  \
+-d name="Kotlin Spring Boot" \
+-d description="Demo project of a Kotlin Spring Boot application" \
+-d packageName=com.okta.developer \
+-o kotlin-spring-boot.zip
+```
+Unzip the file:
+
+```txt
+unzip kotlin-spring-boot.zip -d kotlin-spring-boot
+cd kotlin-spring-boot
+```
+
+## Secure your Application with OpenID Connect
+If you already have an Okta account, see the Create a Web Application in Okta sidebar below. Otherwise, we created a Maven plugin that configures a free Okta developer account + an OIDC app (in under a minute!).
+
+```txt
+./mvnw com.okta:okta-maven-plugin:setup
+```
+
+You should see the following output:
+
+```
+First name: Jimena
+Last name: Garbarino
+Email address: ***
+Company: ***
+Creating new Okta Organization, this may take a minute:
+OrgUrl: https://dev-123456.okta.com
+Check your email address to verify your account.
+
+Writing Okta SDK config to: /home/indiepopart/.okta/okta.yaml
+
+Configuring a new OIDC, almost done:
+Created OIDC application, client-id: ***
+```
+
+Check your email and follow the instructions to activate your Okta account.
+
+> **If you already have an Okta Developer account**
+> 
+> Log in and create a new Application:
+> - From the Applications page, choose Add Application.
+> - On the Create New Application page, select Web.
+> - Give your app a memorable name, add http://localhost:8080/login/oauth2/code/okta and http://localhost:8080/authorization-code/callback as a Login redirect URIs.
+> 
+> Copy the issuer (found under **API** > **Authorization Servers**), client ID, and client secret into `src/main/resources/application.properties`.
+> 
+```properties
+okta.oauth2.issuer=$issuer
+okta.oauth2.client-id=$clientId
+okta.oauth2.client-secret=$clientSecret
+```
+
+Edit `KotlinSpringBootApplication` located the package `com.okta.developer` to add a controller mapping that will print a welcome message in the browser window.
+
+```kotlin
+package com.okta.developer
+
+import org.springframework.boot.autoconfigure.SpringBootApplication
+import org.springframework.boot.runApplication
+import org.springframework.security.core.annotation.AuthenticationPrincipal
+import org.springframework.security.oauth2.client.authentication.OAuth2AuthenticationToken
+import org.springframework.web.bind.annotation.GetMapping
+import org.springframework.web.bind.annotation.RestController
+
+@SpringBootApplication
+@RestController
+class KotlinSpringBootApplication {
+
+    @GetMapping("/")
+    fun hello(@AuthenticationPrincipal authenticationToken: OAuth2AuthenticationToken): String {
+
```

Comment:
Remove blank line.

Reply:
Another way to do this is to grab the `OidcUser` and call `getFullName()`. Not sure if that's worth mentioning.